addons/web/static/src/js/views/basic/basic_model.js

```diff
@@ -1850,7 +1850,9 @@ var BasicModel = AbstractModel.extend({
         var view = fieldInfo.views && fieldInfo.views[fieldInfo.mode];
         var def, rec;
         var defs = [];
-        list._changes = list._changes || [];
+        if (_.isEmpty(list._changes)) {
```

Comment:
`list._changes` can never be an object. If it is the case, this is an issue and that's what should be fixed.
`changes` is an object for datapoints of type record (keys are field names), and an array for datapoints of type list.

Reply:
@aab-odoo As I said it is never produced in normal case, it is too specific case which is produced when we have o2m inside o2m, and we edit inner o2m and save it and then discard main record and again edit and edit inner o2m.

Reply:
@msh-odoo My point is, whatever the scenario, `list._changes` should not be an object. The correct fix here is to prevent it from being set to an object (I don't know where it is done).

Reply:
@aab-odoo Thanks for the review, I changed the fix again, would you please check now?